src/Sylius/Component/Registry/ServiceRegistry.php

```diff
@@ -62,54 +56,54 @@ public function all()
     /**
      * {@inheritdoc}
      */
-    public function register($type, $service)
+    public function register($identifier, $service)
     {
-        if ($this->has($type)) {
-            throw new ExistingServiceException($this->context, $type);
+        if ($this->has($identifier)) {
+            throw new ExistingServiceException($this->context, $identifier);
         }
 
         if (!is_object($service)) {
-            throw new \InvalidArgumentException(sprintf('%s needs to be an object, %s given.', $this->context, gettype($service)));
+            throw new \InvalidArgumentException(sprintf('%s needs to be an object, %s given.', ucfirst($this->context), gettype($service)));
         }
 
-        if (!in_array($this->interface, class_implements($service))) {
+        if (!in_array($this->interface, class_implements($service), true)) {
```

Comment:
Added `true` in order to have `in_array` perform strict comparison rather than the loose one.